### PR TITLE
Add lint script and fix lint errors

### PR DESCRIPTION
**Pull request by RyanZim**
**I would like someone else to review & merge this.**

# Changes:

- Fixed code in `lib/utils.js`, `examples/`, and the `Jakefile` to comply with the eslint rules.
  - Removed unneeded `var fs = require('fs')` in `Jakefile`
- Added `.eslintignore` file
- Set `"avoidEscape": true` in `.eslintrc.json` to allow for using double quotes when the string contains single quotes. See https://github.com/mde/ejs/blob/eslint/lib/utils.js#L52

---

@mde @TimothyGu Should we consider making linting part of the travis build?

### CODE REVIEW

**Comment by RyanZim** [comment]
@mde In light of https://github.com/mde/ejs/pull/198#issuecomment-242108532 and similar, perhaps we should also add http://eslint.org/docs/rules/no-trailing-spaces.

**Comment by RyanZim** [comment]
@mde @TimothyGu ping?

File: Jakefile [hunk 1]
```diff
@@ -1,8 +1,8 @@
-var fs = require('fs')
-  , buildOpts = {
-      printStdout: true
-    , printStderr: true
-    };
+/* global jake, task, complete, desc, publishTask */
+var buildOpts = {
+    printStdout: true,
+    printStderr: true
+  };
```
Comment (TimothyGu) [inline Jakefile]: The indentation should be adjusted. Ditto for other hunks.

Comment (RyanZim) [inline Jakefile]: @TimothyGu Do you think we should add http://eslint.org/docs/rules/indent?

**Comment by RyanZim** [comment]
@TimothyGu Fixed and added indent rule. Anything else?

File: examples/functions.js [hunk 2]
```diff
@@ -2,17 +2,17 @@
...
-    };
+var ejs = require('../');
+var read = require('fs').readFileSync;
+var join = require('path').join;
+var path = join(__dirname, '/functions.ejs');
+var data = {
+  users: [
+        { name: 'Tobi', age: 2, species: 'ferret' },
+        { name: 'Loki', age: 2, species: 'ferret' },
+        { name: 'Jane', age: 6, species: 'ferret' }
```
Comment (TimothyGu) [inline examples/functions.js]: Here too

**Comment by RyanZim** [comment]
@TimothyGu Fixed. Anything else?

**Comment by RyanZim** [comment]
@mde @TimothyGu ping?

**Comment by mde** [comment]
I am down with stripping the trailing spaces. I'll review this ASAP.

**Comment by RyanZim** [comment]
OK, thoughts on `no-trailing-spaces`?

**Comment by mde** [comment]
That's what I said "I am down with stripping trailing spaces."

**Comment by mde** [comment]
LGTM

**Pull request merged**